src/System.Private.CoreLib/src/System/Attribute.CoreRT.cs

```diff
@@ -123,12 +54,6 @@ public override int GetHashCode()
             return type.GetHashCode();
         }
 
-        public virtual object TypeId => GetType();
-
-        public virtual bool Match(object obj) => Equals(obj);
-
-        public virtual bool IsDefaultAttribute() => false;
-
         //
         // This non-contract method is known to the IL transformer. See comments around _ILT_ReadFields() for more detail.
```

Comment:
@MichalStrehovsky It does not look like we have these special methods implemented in the CoreRT. I do not think we need them - they look like an complex ProjectN-specific micro-optimization. Should we keep this for PROJECTN only and keep the reflection path for CoreRT as well?

Reply:
I'm only working from my phone now so I can't check, but I think many of our FX defined attributes are going to have private fields and we don't generate metadata for them (they are not going to show up in the field list). I don't think we're going to lift the "framework-private stuff is invisible to reflection" policy for our default configurations anytime soon because this is a nice size optimization.

The RtFieldInfo stuff is not going to work on CoreRT anyway because it's the kind of implementation detail I wouldn't expect us to share.

Reply:
> framework-private stuff is invisible to reflection

IMHO, this unconditional policy is not good. It required us to implement many one-off creative workarounds in .NET Native. And we gave up in .NET Native at the end anyway and introduced exceptions for it for binary serialization. I think we should be open to introducing exceptions for it as necessary (like for attribute fields here), so that we do not have to be coming up with these creative workarounds.

> The RtFieldInfo stuff is not going to work on CoreRT anyway

I have cleaned it up in CoreCLR. These special Unsafe getters/setters were not really necessary. I will wait with merging this one till the CoreCLR cleanup goes through.

Reply:
Right, and as we discussed when we were making the holes for binary serialization, punching holes through reflection blocking requires some good tooling on the buildtools/arcade side so that we don't end up in situations like:

```csharp
internal class MyBlockedType { }

public class MyAttribute : Attribute
{
    private MyBlockedType _foo;
}
```

If we punch a hole through blocking for the `_foo` field, we are in a situation where we need to make sure `MyBlockedType` also has metadata enabled for it (it's marked `internal`, so it would be blocked). It needs to be a build break at _managed code compilation type_ to enforce this.

If we don't make it fail at managed code compilation time, this is going to be either an internal compiler error in ILC, or a runtime failure to reflect over the type. Neither of which is a good experience (or a reliable way to catch this bug before customers hit it in some obscure scenario we don't test).